src/layers/legacy/medCoreLegacy/database/medAbstractDatabaseImporter.cpp

```diff
@@ -327,82 +345,86 @@ void medAbstractDatabaseImporter::importFile ( void )
+                // 3.3) a) re-populate missing metadata
+                // as files might be aggregated we use the aggregated file name as SeriesDescription (if not provided, of course)
+                QFileInfo imagefileInfo(filesPaths[0]);
+                patientID = imagesGroupedByPatient[aggregatedFileName];
+                QString seriesID = imagesGroupedBySeriesId[aggregatedFileName];
+
+                populateMissingMetadata(medData, med::smartBaseName(imagefileInfo.fileName()));
+                medData->setMetaData(medMetaDataKeys::PatientID.key(), QStringList() << patientID);
+                medData->setMetaData(medMetaDataKeys::SeriesID.key(), QStringList() << seriesID);
+
+                // 3.3) b) now we are able to add some more metadata
+                addAdditionalMetaData (medData, aggregatedFileName, filesPaths);
+            }
+            else
+            {
+                qWarning() << "Could not repopulate data!";
+                emit showError (tr("Could not read data: ") + filesPaths[0], 5000);
+                emit dataImported(medDataIndex(), d->uuid);
+                emit failure(this);
+                return;
             }
 
-            // now writing file
-            bool writeSuccess = tryWriteImage ( fileInfo.filePath(), imagemedData );
-
-            if ( !writeSuccess )
+            if (!d->indexWithoutImporting)
             {
-                emit showError (tr ( "Could not save data file: " ) + filesPaths[0], 5000 );
-                continue;
+                // create location to store file
+                QFileInfo fileInfo(medStorage::dataLocation() + aggregatedFileName);
+                if (!fileInfo.dir().exists() && !medStorage::mkpath(fileInfo.dir().path()))
+                {
+                    qDebug() << "Cannot create directory: " << fileInfo.dir().path();
+                    continue;
+                }
+
+                // now writing file
+                bool writeSuccess = tryWriteImage(fileInfo.filePath(), medData);
+
+                if (!writeSuccess)
+                {
+                    emit showError(tr("Could not save data file: ") + filesPaths[0], 5000);
+                    continue;
+                }
             }
-        }
-        atLeastOneImportSucceeded = true;
+            atLeastOneImportSucceeded = true;
 
-        // and finally we populate the database
-        QFileInfo aggregatedFileNameFileInfo ( aggregatedFileName );
-        QString pathToStoreThumbnails = aggregatedFileNameFileInfo.dir().path() + "/" + aggregatedFileNameFileInfo.completeBaseName() + "/";
-        index = this->populateDatabaseAndGenerateThumbnails ( imagemedData, pathToStoreThumbnails );
+            // and finally we populate the database
+            QFileInfo aggregatedFileNameFileInfo(aggregatedFileName);
+            QString pathToStoreThumbnails = aggregatedFileNameFileInfo.dir().path() + "/" + aggregatedFileNameFileInfo.completeBaseName() + "/";
```

Comment:
Isn't it better to use QDir::separator() instead of '/'?

Reply:
Indeed. I'm suprised it works on all platforms.